src/tools/rustfmt/src/utils.rs

```diff
@@ -132,6 +132,19 @@ pub(crate) fn format_mutability(mutability: ast::Mutability) -> &'static str {
     }
 }
 
+#[inline]
+pub(crate) fn format_pin_and_mut(
+    pinnedness: ast::Pinnedness,
+    mutability: ast::Mutability,
+) -> &'static str {
+    match (pinnedness, mutability) {
+        (ast::Pinnedness::Not, ast::Mutability::Mut) => "mut ",
+        (ast::Pinnedness::Not, ast::Mutability::Not) => "",
+        (ast::Pinnedness::Pinned, ast::Mutability::Mut) => "pin mut ",
+        (ast::Pinnedness::Pinned, ast::Mutability::Not) => "pin const ",
```

Comment:
Not something that we need to address right now, but it's possible that someone leaves a comment between the keywords. e.g `pin /* comment */ mut`.

Reply:
I separated these two prefixes and now it can handle the comment between `pin` and `mut` (or `const`).